### ISSUE TITLE
Markup issue: associated type may not live long enough

### ISSUE THREAD

**Issue by svieujot** [leptos-rs/leptos#3380]
The following markup works fine:
```
<Show
    when=show_tripped
    fallback=|| {
        view! {
            <span title="ON">
                <Icon icon={i::FaPowerOffSolid} {..} class=css::onIcon/>
            </span>
        }
    }
>
    <span title="Tripped" class=css::tripped>
        <Icon icon={i::TbCircuitSwitchOpen} />
    </span>
</Show>
```
But when moving the css::tripped after the `{..}` operator on the second `<Icon/>` tag like:
```
<Show
    when=show_tripped
    fallback=|| {
        view! {
            <span title="ON">
                <Icon icon={i::FaPowerOffSolid} {..} class=css::onIcon/>
            </span>
        }
    }
>
    <span title="Tripped" >
        <Icon icon={i::TbCircuitSwitchOpen} {..} class=css::tripped/>
    </span>
</Show>
```
Then the markup generates this error:
```

error[E0310]: the associated type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(Class<&str>,)>` may not live long enough
   --> cmd/ui/src/components/propulsion/on_buttons.rs:135:5
    |
135 | /     view! {
136 | |             <Show when=move || show_on_button() || show_start_button() || show_throttle_warning()>
137 | |                 <Show when=show_hydraulic_pump_button>
138 | |                     <div class=classes!(css::hydraulicPump, "text-center")>
...   |
211 | |             </Show>
212 | |         }
    | |         ^
    | |         |
    | |_________the associated type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(Class<&str>,)>` must be valid for the static lifetime...
    |           ...so that the type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(Class<&str>,)>` will meet its required lifetime bounds
    |
    = help: consider adding an explicit lifetime bound `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(Class<&str>,)>: 'static`...
    = note: this error originates in the macro `view` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by gbj** [comment]
My assumption would be that at least one of these values (`css::tripped`, for example) is a borrowed, not an owned, type, and needs to be owned. For example, if `css::tripped` is `&str`, then it needs to be either a `String` or `&'static str`, etc.

If you provide a reproducible example (i.e., fill in the missing parts in the examples above), I'm happy to look at it in more depth.

**Comment by svieujot** [comment]
`css::tripped` is a `&str` generated by stylance.

I will prepare a full example.

**Comment by opensourcecheemsburgers** [comment]
Hey Greg, here's a repro: https://github.com/opensourcecheemsburgers/markup_issue

**Comment by mscofield0** [comment]
I have a similar issue, with the same error.
```rust
// with wu::ShowOption being:
#[component]
pub fn ShowOption<T, VF, IV, TF>(
	/// The data to process.
	data: TF,
	/// A closure that returns what gets rendered if the data is `None`. By default this is the empty view.
	#[prop(optional, into)]
	fallback: ViewFn,
	/// The children will be shown whenever the data is `Some`.
	children: VF,
) -> impl IntoView
where
	VF: Send + Sync + Fn(T) -> IV + 'static,
	T: Send + Sync + Clone + 'static,
	IV: IntoView + 'static,
	TF: Send + Sync + Fn() -> Option<T> + 'static,
{
	let data = Signal::derive(data);
	move || match data.get() {
		Some(data) => Either::Left(children(data)),
		None => Either::Right(fallback.clone().run()),
	}
}


// the code in question
<wu::ShowOption
    data=move || state.get().session().get()
    fallback=move || view! {
        <div class="vertical gap-2">
            <A href="/login" {..} class="flex hcenter link opacity-75 aria-[current=page]:opacity-100">{tr_login}</A>
            <A href="/register" {..} class="flex hcenter link opacity-75 aria-[current=page]:opacity-100">{tr_register}</A>
            <div class="horizontal gap-4">
                <button class="btn-circle" on:click=move |_| toggle_app_settings.set(true)>
                    <span class="i-o-cog-6"/>
                </button>
                <A href="/help" {..} class="btn-circle">
                    <span class="i-o-question-mark-circle" />
                </A>
            </div>
        </div>
    }
    let:session
>
...
</wu::ShowOption>
```
```rust
error[E0310]: the associated type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(Class<&str>,)>` may not live long enough
   --> frontend/src/app/routes/mod.rs:232:2
    |
232 | /     view! {
233 | |         <AppSettings toggle=toggle_app_settings />
234 | |
235 | |         <wu::Drawer
...   |
325 | |         </wu::Drawer>
326 | |     }
    | |     ^
    | |     |
    | |_____the associated type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(Class<&str>,)>` must be valid for the static lifetime...
    |       ...so that the type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(Class<&str>,)>` will meet its required lifetime bounds
    |
    = help: consider adding an explicit lifetime bound `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(Class<&str>,)>: 'static`...
    = note: this error originates in the macro `view` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by gbj** [comment]
At this point, I'm somewhat confused -- I cannot reproduce an error with the reproduction provided by @opensourcecheemsburgers, which compiles and runs (see below)
![Screenshot 2024-12-19 at 3 57 06 PM](https://github.com/user-attachments/assets/97525fe6-9c3b-4bfd-9655-62aee8f9b3d0)

And likewise I can't reproduce an error with your example @mscofield0 (which I tried against the current main branch). I had to fill out a bit more of the reproduction for this to compile:

<details>
<summary>The rest of the code I tried</summary>
```rust
use leptos::{either::Either, prelude::*};
use leptos_router::components::A;

// with wu::ShowOption being:
#[component]
pub fn ShowOption<T, VF, IV, TF>(
    /// The data to process.
    data: TF,
    /// A closure that returns what gets rendered if the data is `None`. By default this is the empty view.
    #[prop(optional, into)]
    fallback: ViewFn,
    /// The children will be shown whenever the data is `Some`.
    children: VF,
) -> impl IntoView
where
    VF: Send + Sync + Fn(T) -> IV + 'static,
    T: Send + Sync + Clone + 'static,
    IV: IntoView + 'static,
    TF: Send + Sync + Fn() -> Option<T> + 'static,
{
    let data = Signal::derive(data);
    move || match data.get() {
        Some(data) => Either::Left(children(data)),
        None => Either::Right(fallback.clone().run()),
    }
}

#[component]
pub fn App() -> impl IntoView {
    view! {
    // the code in question
    <ShowOption
        data=move || Some(String::from("foo"))
        fallback=move || view! {
            <div class="vertical gap-2">
                <A href="/login" {..} class="flex hcenter link opacity-75 aria-[current=page]:opacity-100">"Login"</A>
                <A href="/register" {..} class="flex hcenter link opacity-75 aria-[current=page]:opacity-100">"Register"</A>
                <div class="horizontal gap-4">
                    <button class="btn-circle">
                        <span class="i-o-cog-6"/>
                    </button>
                    <A href="/help" {..} class="btn-circle">
                        <span class="i-o-question-mark-circle" />
                    </A>
                </div>
            </div>
        }
        let:session
    >
        "bar"
    </ShowOption>
    }
}

fn main() {
    leptos::mount::mount_to_body(|| view! { <App /> });
}
```
</details>

I had been using an older nightly compiler, and wondered if that was the problem, so I switched to the latest and -- still no errors.

Not sure what to say at this point. Maybe share your `rustc --version`?

**Comment by adrianncovaci** [comment]
I can confirm both code snippets compile fine with the latest rustc:
```
nightly-x86_64-unknown-linux-gnu (default)
rustc 1.85.0-nightly (9e136a30a 2024-12-19)
```

**Comment by svieujot** [comment]
Indeed a `rustup update` fixed the problem.

Thank you Adrian.

**Comment by mscofield0** [comment]
I'm using stable rustc @gbj

**Comment by gbj** [comment]
@mscofield0 Just now, I tried with `rustc 1.81.0 (eeb90cda1 2024-09-04)`, which was the stable I happened to have installed, then ran `rustup update stable` and tried with `rustc 1.83.0 (90b35a623 2024-11-26)` — still no error.

Like I mentioned, I had to fill in a few details to make your snippet compile. If you have a complete, self-contained example that errors, feel free to share it and I'll try again.
